frontend/src/app/pages/category-list/category-list.component.html

```diff
@@ -1,42 +1,44 @@
-<ng-container *ngIf="!categories">
-  <mat-progress-spinner
-    class="example-margin"
-    color="primary"
-    mode="indeterminate"
-  >
-  </mat-progress-spinner>
+<ng-container *ngIf="this.loading">
+  <div class="spinner">
+    <mat-progress-spinner
+      class="example-margin"
+      color="primary"
+      mode="indeterminate"
+    >
+    </mat-progress-spinner>
+  </div>
 </ng-container>
 
-<ng-container *ngIf="role && categories">
+<ng-container *ngIf="!this.loading">
   <div class="container">
-    <h1 style="color: black">Category database</h1>
+    <h1 style="color: black">Category table</h1>
     <div class="row">
     </div>
 
     <div class="row">
-      <table mat-table [dataSource]="categories" class="mat-elevation-z8">
+      <table mat-table [dataSource]="dataSource" class="mat-elevation-z8" matSort (matSortChange)="sortChanged($event)">
 
         <ng-container matColumnDef="id">
-          <th mat-header-cell *matHeaderCellDef> #</th>
+          <th mat-header-cell *matHeaderCellDef mat-sort-header = 'id'> #</th>
           <td mat-cell *matCellDef="let category"> {{category.id}} </td>
         </ng-container>
 
         <ng-container matColumnDef="name">
-          <th mat-header-cell *matHeaderCellDef> Name</th>
+          <th mat-header-cell *matHeaderCellDef mat-sort-header = 'name'> Name</th>
           <td mat-cell *matCellDef="let category"> {{category.name}} </td>
         </ng-container>
 
         <ng-container matColumnDef="description">
-          <th mat-header-cell *matHeaderCellDef> Description</th>
-          <td mat-cell *matCellDef="let category"> {{category.description}} </td>
+          <th style="width: 800px" mat-header-cell *matHeaderCellDef mat-sort-header = 'description'> Description</th>
```

Comment:
`width: 800px` is bad practice, because we have small screens in phones. Use percentages